afpacket/src/tokio_sockets.rs

```diff
@@ -20,11 +35,287 @@ impl AsyncBoundSocket {
         self.sock.get_mut().set_promiscuous(p)
     }
 
-    pub async fn send(&mut self, frame: &[u8]) -> io::Result<usize> {
-        self.sock.write(frame).await
+    pub fn split(self) -> (SendHalf, RecvHalf) {
+        let inner = Arc::new(Inner {
+            locked: AtomicBool::new(false),
+            sock: UnsafeCell::new(self),
+        });
+
+        let rx = RecvHalf {
+            inner: inner.clone(),
+        };
+        let tx = SendHalf { inner };
+
+        (tx, rx)
+    }
+
+    pub fn poll_can_rx(&mut self, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
```

Comment:
Docs: When does this return Ready, when does it return Pending.  Add something for each of these poll_* functions. 
Should also specify that these functions are non-blocking.  Just think of all the things you want to know when you work with a library for functions like these.